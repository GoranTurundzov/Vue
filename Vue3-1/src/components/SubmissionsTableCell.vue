<template>
    <div :class="direction">
        <div class="title text-capitalize">{{ title.replace('_', ' ') }}</div>
        <div class="font-w600 data-content" :class="title">{{ text }}</div>
    </div>
</template>

<script>
export default {
    name: 'SubmissionsTableRowColumn',
    props: {
        title: {
            required: true,
            description: 'The title of a single column from a table',
            type: String
        },
        text: {
            required: true,
            description: 'The content of a single column from a table',
            type: [String, Number, Boolean]
        },
        alignInRows: {
            type: Boolean,
            description: 'Column or Row that will affect how the cells will allign',
            default: false
        }
    },
    computed: {
        direction() {
            if (this.alignInRows == true) {
                return 'row border-b';
            }
            return 'column';
        }
    }
};
</script>
