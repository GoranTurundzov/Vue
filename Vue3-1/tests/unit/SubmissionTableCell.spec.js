import { mount } from '@vue/test-utils';
import SubmissionsTableCell from '@/components/SubmissionsTableCell';

describe('SubmissionsTableCell', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = mount(SubmissionsTableCell, {
			props: {
				title: 'name',
				text: 'Goran',
				alignInRows: false
			}
		});
	});

	it('Matches snapshot', () => expect(wrapper.element).toMatchSnapshot());

	it('Accepts props', () => {
		let title = wrapper.find('.title');
		let content = wrapper.find('.data-content');

		expect(title.text()).toEqual('name');
		expect(content.text()).toEqual('Goran');
	});
});
