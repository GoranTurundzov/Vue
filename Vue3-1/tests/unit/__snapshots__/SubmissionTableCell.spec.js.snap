// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmissionsTableCell Matches snapshot 1`] = `
<div
  class="column"
>
  <div
    class="title text-capitalize"
  >
    name
  </div>
  <div
    class="font-w600 data-content name"
  >
    Goran
  </div>
</div>
`;
